schema/collection.go

```diff
@@ -122,7 +122,9 @@ func disableAdditionalPropertiesAndAllowNullable(required []string, properties m
 		}
 
 		if len(p.Properties) > 0 {
-			p.AdditionalProperties = false
+			if p.AdditionalProperties == nil {
```

Comment:
What about honoring this at line `147` as well? Then if it is passed we honor it top level as well?